pyEpiabm/pyEpiabm/core/household.py

```diff
@@ -104,3 +106,26 @@ def remove_household(self):
         """
         self.microcell.households.remove(self)
         self.cell.households.remove(self)
+
+    def set_id(self, id: str):
+        """Updates ID of household (i.e. for input from file).
+        Format of ID - for example 3.1.2 represents household 2 within
+        microcell 1 within cell 3. The id will only be changed if it has the
+        correct expression.
+
+        Parameters
+        ----------
+        id : str
+            Identity of household
+
+        """
+
+        # Ensure id is a string
+        if not isinstance(id, str):
+            raise TypeError("id must be of type string")
```

Comment:
```suggestion
            raise TypeError("Provided id must be a string")
```

I think this phrasing is more natural - change throughout